File: TYPESCRIPT/Cap07/c07ex03.ts
// c07ex03.ts

import * as rls from 'readline-sync';

let A: number, B: number;

function TROCA(): void {
  let X: number = A;
  A = B;
  B = X;
}

A = rls.questionInt("Entre um valor para a variavel <A>: ");
B = rls.questionInt("Entre um valor para a variavel <B>: ");

TROCA();

console.log("");
console.log(`Variavel <A> com valor ${A}`);
console.log(`Variavel <B> com valor ${B}`);

console.log("");
process.stdout.write("Tecle <Enter> para encerrar... ");
rls.question();

File: GO/Cap07/c07ex03.go
// c07ex03.go

package main
import "fmt"

var A, B int

func TROCA() {
  var X int
  X = A
  A = B
  B = X
}

func main() {
  fmt.Print("Entre um valor para a variavel <A>: ")
  fmt.Scan(&A)

  fmt.Print("Entre um valor para a variavel <B>: ")
  fmt.Scan(&B)

  TROCA()
  
  fmt.Println()
  fmt.Println("Variavel <A> com valor", A)
  fmt.Println("Variavel <B> com valor", B)

  fmt.Println()
  fmt.Print("Tecle <Enter> para encerrar... ")
  fmt.Scanln()
}
